src/mbgl/renderer/render_layer.cpp

```diff
@@ -62,4 +64,27 @@ bool RenderLayer::needsRendering(float zoom) const {
            && baseImpl->maxZoom >= zoom;
 }
 
-} // namespace mbgl
+void RenderLayer::setRenderTiles(std::vector<std::reference_wrapper<RenderTile>> tiles) {
```

Comment:
I think we'll eventually want to push `setRenderTiles`, `uploadBuckets`, and `render` down into individual subclasses, even it if means duplicating a bit of code. If we push the loop over tiles down, then that opens up an opportunity to pull loop invariant rendering code out of the loop, improving performance.

This doesn't have to be done immediately though.